### ISSUE TITLE
🐛 [cloud_firestore] NullPointerException in FlutterFirebaseFirestorePlugin.removeEventListeners

### ISSUE THREAD

**Issue by kinex** [FirebaseExtended/flutterfire#3670]
## Bug report

I see this from Crashlytics:
```
Fatal Exception: java.lang.RuntimeException
Unable to destroy activity {com.***/com.***.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void c.f.e.t.z.remove()' on a null object reference

Caused by java.lang.NullPointerException
Attempt to invoke interface method 'void c.f.e.t.z.remove()' on a null object reference
io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin.removeEventListeners

Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void c.f.e.t.z.remove()' on a null object reference
       at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin.removeEventListeners(FlutterFirebaseFirestorePlugin.java:23)
       at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin.onDetachedFromEngine(FlutterFirebaseFirestorePlugin.java)
       at io.flutter.embedding.engine.FlutterEnginePluginRegistry.remove(FlutterEnginePluginRegistry.java:118)
       at io.flutter.embedding.engine.FlutterEnginePluginRegistry.remove(FlutterEnginePluginRegistry.java:16)
       at io.flutter.embedding.engine.FlutterEnginePluginRegistry.removeAll(FlutterEnginePluginRegistry.java:11)
       at io.flutter.embedding.engine.FlutterEnginePluginRegistry.destroy(FlutterEnginePluginRegistry.java:10)
       at io.flutter.embedding.engine.FlutterEngine.destroy(FlutterEngine.java:9)
       at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(FlutterActivityAndFragmentDelegate.java:90)
       at io.flutter.embedding.android.FlutterActivity.onDestroy(FlutterActivity.java:10)
       at android.app.Activity.performDestroy(Activity.java:8225)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1342)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5441)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5493)
       at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:44)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2267)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:8154)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)

```

Flutter doctor:
[√] Flutter (Channel stable, 1.20.4, on Microsoft Windows [Version 10.0.18363.1082], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[√] Android Studio (version 3.5)
[√] VS Code (version 1.49.2)
[√] Connected device (1 available)

cloud_firestore: "^0.14.0+2"

**Comment by itssidhere** [comment]
Related #3535

**Comment by kinex** [comment]
Callstack looks different.

**Comment by itssidhere** [comment]
This is fixed already in this PR #3452 until it lands in production use this [https://github.com/FirebaseExtended/flutterfire/issues/3354#issuecomment-683241775](url)

**Comment by russellwheatley** [comment]
I believe this problem has been solved, at least, #3452 has now been merged.
